registration/include/pcl/registration/icp.h

```diff
@@ -321,17 +324,20 @@ namespace pcl
   template <typename PointSource, typename PointTarget, typename Scalar = float>
   class IterativeClosestPointWithNormals : public IterativeClosestPoint<PointSource, PointTarget, Scalar>
   {
+    using Self = IterativeClosestPointWithNormals<PointSource, PointTarget, Scalar>;
+    using Base = IterativeClosestPoint<PointSource, PointTarget, Scalar>;
+    
     public:
-      using PointCloudSource = typename IterativeClosestPoint<PointSource, PointTarget, Scalar>::PointCloudSource;
-      using PointCloudTarget = typename IterativeClosestPoint<PointSource, PointTarget, Scalar>::PointCloudTarget;
-      using Matrix4 = typename IterativeClosestPoint<PointSource, PointTarget, Scalar>::Matrix4;
+      using PointCloudSource = typename Self::PointCloudSource;
```

Comment:
Also these 3 should be base.

Reply:
Sry I even failed search & replace. Let me update this.